### ISSUE TITLE
KeyError: 'product' when ducling handling keyword 'meat'

### ISSUE THREAD

**Issue by DiWuRR** [FraBle/python-duckling#30]
I have following problem when I use duckling to deal with time in RASA NLU. It only happens when the text contain 'meat', like 'gilled meat'. but it is fine with 'Meat' or 'gilledmeat'. 

---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-17-4d602aada7e5> in <module>()
      1 # Test Block for RASA NLU Performance
      2 Sentence="gilled meat"
----> 3 interpreter.parse(Sentence.lower())

/opt/app/anaconda3/lib/python3.6/site-packages/rasa_nlu/model.py in parse(self, text, time)
    293 
    294         for component in self.pipeline:
--> 295             component.process(message, **self.context)
    296 
    297         output = self.default_output_attributes()

/opt/app/anaconda3/lib/python3.6/site-packages/rasa_nlu/extractors/duckling_extractor.py in process(self, message, **kwargs)
    128                                 "".format(message.time, ref_time, e))
    129 
--> 130         matches = self.duckling.parse(message.text, reference_time=ref_time)
    131         relevant_matches = [match
    132                             for match in matches

/opt/app/anaconda3/lib/python3.6/site-packages/duckling/wrapper.py in parse(self, input_str, reference_time)
    236             ]
    237         """
--> 238         return self._parse(input_str, reference_time=reference_time)
    239 
    240     def parse_time(self, input_str, reference_time=''):

/opt/app/anaconda3/lib/python3.6/site-packages/duckling/wrapper.py in _parse(self, input_str, dim, reference_time)
     54         duckling_result = self.duckling.parse(
     55             input_str, self.language, dim_filter=dim,
---> 56             reference_time=reference_time)
     57         for entry in duckling_result:
     58             if entry[u'dim'] in self._dims:

/opt/app/anaconda3/lib/python3.6/site-packages/duckling/duckling.py in parse(self, input_str, language, dim_filter, reference_time)
    159                 language, input_str, self.clojure.read(filter_str))
    160 
--> 161         return self._parse_result(duckling_result)
    162 
    163     def _parse_reference_time(self, reference_time):

/opt/app/anaconda3/lib/python3.6/site-packages/duckling/duckling.py in _parse_result(self, duckling_result)
    185                 if key == u'value':
    186                     entry[key] = self._parse_dict(
--> 187                         field.getValue(), entry[u'dim'])
    188                 else:
    189                     entry[key] = _functions[key](field.getValue())

/opt/app/anaconda3/lib/python3.6/site-packages/duckling/duckling.py in _parse_dict(self, java_dict, dim)
    220                 result[key] = _functions_with_dim[key](field.getValue(), dim)
    221             else:
--> 222                 result[key] = _functions[key](field.getValue())
    223         return result
    224 

KeyError: 'product'
---------------------------------------------------------------------------

BTW, below are the version of dependency packages: 
1.       Duckling: 1.7.3
2.       Spacy: 2.0.3
3.       Sklearn: 0.19.0
4.       rasa_nlu: 0.10.6

Configuration of RASA as follow: 
{
  "pipeline": ["nlp_spacy", "tokenizer_spacy", "intent_entity_featurizer_regex", "intent_featurizer_spacy", "ner_crf","ner_spacy", "ner_synonyms",  "intent_classifier_sklearn","ner_duckling"],
  "path" : "/*******************/models",
  "data" : "/*******************/JSON_File.json",
      "fine_tune_spacy_ner": true
}

Many thanks!

**Comment by careless25** [comment]
Any update on this? We are also facing this issue. Same versions of all the dependencies as above

**Comment by tbrooks007** [comment]
We are having the same KeyError: 'product' but our magic word that raises the exception is "sugar" 

**Dependency Packages:**
duckling==1.7.3
spacy==1.9.0
scikit-learn==0.19.0
rasa-nlu==0.10.1

**Rasa Configuration:**
{
  "project": "detector",
  "language": "en",
  "pipeline": ["nlp_spacy", "tokenizer_spacy", "intent_entity_featurizer_regex", "intent_featurizer_spacy", "ner_crf",
               "ner_duckling", "ner_spacy", "ner_synonyms", "intent_classifier_sklearn"],
  "path" : "models/",
  "data" : "data/train.json",
  "response_log": "logs/",
  "duckling_dimensions": ["duration", "time"],
  "intent_classifier_sklearn": {
    "C": [1, 2, 5, 10, 20, 100],
    "kernel": "linear"
  }
}

**Partial Stacktrace**
```
  File ".../lib/python3.6/site-packages/duckling/duckling.py", line 222, in _parse_dict
    result[key] = _functions[key](field.getValue())
KeyError: 'product'
```

**Example Sentences**
- Maltitol is an alcohol sugar and can be undigestible in the body.
- The brown sugar and chocolate mask any bean taste that other reviews have mentioned.
- Substitute this to sugar and you'll be doing your body a great favor.

**Comment by FraBle** [comment]
Hey guys, I will look into this, an update will follow shortly!

**Comment by FraBle** [comment]
It should be fixed now with [`duckling v1.8.0`](https://pypi.python.org/pypi/duckling/1.8.0).
